tests/robustness/checker/checker.go

```diff
@@ -0,0 +1,275 @@
+// Package checker defines the framework for creating and restoring snapshots
+// with a data integrity check
+package checker
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"github.com/kopia/kopia/tests/robustness/snapif"
+	"github.com/kopia/kopia/tests/robustness/snapstore"
+)
+
+// Checker is an object that can take snapshots and restore them, performing
+// a validation for data consistency
+type Checker struct {
+	RestoreDir string
+	snap       snapif.Snapshotter
+	snapStore  snapstore.Storer
+	validator  Comparer
+}
+
+// NewChecker instantiates a new Checker, returning its pointer. A temporary
+// directory is created to mount restored data
+func NewChecker(snap snapif.Snapshotter, snapStore snapstore.Storer, validator Comparer) (*Checker, error) {
+	restoreDir, err := ioutil.TempDir("", "restore-data-")
+	if err != nil {
+		return nil, err
+	}
+
+	return &Checker{
+		RestoreDir: restoreDir,
+		snap:       snap,
+		snapStore:  snapStore,
+		validator:  validator,
+	}, nil
+}
+
+// Cleanup cleans up the Checker's temporary restore data directory
+func (chk *Checker) Cleanup() {
+	if chk.RestoreDir != "" {
+		os.RemoveAll(chk.RestoreDir) //nolint:errcheck
+	}
+}
+
+// GetSnapIDs gets the list of snapshot IDs being tracked by the checker's snapshot store
+func (chk *Checker) GetSnapIDs() []string {
+	return chk.snapStore.GetKeys()
+}
+
+// SnapshotMetadata holds metadata associated with a given snapshot
+type SnapshotMetadata struct {
```

Comment:
it makes sense to add JSON tags to the struct to follow the convention for JSON objects, in particular field names.

Reply:
added